packages/x-date-pickers/src/DateTimeField/useDateTimeField.ts

```diff
@@ -36,8 +36,11 @@ const useDefaultizedDateTimeField = <TDate, AdditionalProps extends {}>(
     disablePast: props.disablePast ?? false,
     disableFuture: props.disableFuture ?? false,
     format: props.format ?? utils.formats.keyboardDateTime,
-    minDate: applyDefaultDate(utils, props.minDate, defaultDates.minDate),
-    maxDate: applyDefaultDate(utils, props.maxDate, defaultDates.maxDate),
+    disableIgnoringDatePartForTimeValidation: Boolean(props.minDateTime || props.maxDateTime),
```

Comment:
For v6 I wanted to remove this prop from the public signature and move the `minTime: props.minDateTime ?? props.minTime` logic inside the validation.

But maybe we can temporarily add it here